test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlCustomQueryTests.cs

```diff
@@ -68,15 +69,23 @@ public async Task GivenASqlQuery_IfAStoredProcExistsWithMatchingHash_ThenStoredP
             _output.WriteLine("Adding new sproc to database.");
             AddSproc(hash);
 
-            await Task.Delay(1100);
-
             // Query after adding an sproc to the database
-            await _fixture.SearchService.SearchAsync(KnownResourceTypes.OrganizationAffiliation, queryParameters, CancellationToken.None);
-
-            Assert.Equal(hash, _fixture.SqlQueryHashCalculator.MostRecentSqlHash);
+            var sw = Stopwatch.StartNew();
+            var sprocWasUsed = false;
+            while (sw.Elapsed.TotalSeconds < 10) // previous single try after 1.1 sec delay was not reliable.
+            {
+                await Task.Delay(300);
+                await _fixture.SearchService.SearchAsync(KnownResourceTypes.OrganizationAffiliation, queryParameters, CancellationToken.None);
+                Assert.Equal(hash, _fixture.SqlQueryHashCalculator.MostRecentSqlHash);
+                if (await CheckIfSprocUsed(hash))
+                {
+                    sprocWasUsed = true;
+                    break;
+                }
+            }
 
             // assert an sproc was not used
```

Comment:
This comment doesn't look accurate. The assert is checking that the sproc was used.